src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -405,10 +405,20 @@ ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
                                        uct_ib_mlx5_mmio_mode_t mmio_mode)
 {
 #if HAVE_DEVX
-    uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context, 0);
+    uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
+                                                UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
     if (uar->uar == NULL) {
-        ucs_error("mlx5dv_devx_alloc_uar() failed: %m");
-        return UCS_ERR_NO_MEMORY;
+        if (UCT_IB_MLX5_UAR_ALLOC_TYPE_WC == UCT_IB_MLX5_UAR_ALLOC_TYPE_NC) {
+            ucs_error("mlx5dv_devx_alloc_uar(WC) failed: %m");
+            return UCS_ERR_NO_MEMORY;
+        }
+
+        uar->uar        = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
+                                                UCT_IB_MLX5_UAR_ALLOC_TYPE_NC);
+        if (uar->uar == NULL) {
+            ucs_error("mlx5dv_devx_alloc_uar(NC) failed: %m");
+            return UCS_ERR_NO_MEMORY;
+        }
```

Comment:
IMO, `#define UCT_IB_MLX5_UAR_ALLOC_TYPE_NC   0` is not correct (but only `#define UCT_IB_MLX5_UAR_ALLOC_TYPE_WC   0` is correct)
maybe add `#ifdef HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC` here?
and check for HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_WC separately in configure?